Add LLVM module support for sitofp instruction

This will be needed to implement #323.

This is part of #262.

Latest log of windows build fail -

```
 Checking Build System
33
  Building Custom Rule D:/a/nballerina-cpp/nballerina-cpp/nBallerina-project/runtime/c_rt/CMakeLists.txt
34
cl : command line error D8021: invalid numeric argument '/Wno-override-module' [D:\a\nballerina-cpp\nballerina-cpp\nBallerina-project\build\runtime\c_rt\ballerina_crt.vcxproj]
35
  Generate static runtime lib
36
  'CARGO_TARGET_DIR' is not recognized as an internal or external command,
37
  operable program or batch file.
38
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(240,5): error MSB8066: Custom build for 'D:\a\nballerina-cpp\nballerina-cpp\nBallerina-project\build\CMakeFiles\7374d569daff70cef39c665e6f968a35\libballerina_rt.rule' exited with code 9009. [D:\a\nballerina-cpp\nballerina-cpp\nBallerina-project\build\runtime\rust_rt\libballerina_rt.vcxproj]
39
Error: Process completed with exit code 1.
```

--- a/compiler/modules/print.llvm/llvm.bal
+++ b/compiler/modules/print.llvm/llvm.bal
@@ -740,6 +740,19 @@ public class Builder {
         }
     }
 
+    // Corresponds to LLVMBuildSIToFP
+    public function sIToFP(Value val, RealType destTy, string? name=()) returns Value {
+        if val.ty is IntType {
+            BasicBlock bb = self.bb();
+            string|Unnamed reg = bb.func.genReg(name);
+            bb.addInsn(reg, "=", "sitofp", typeToString(val.ty), val.operand, "to", typeToString(destTy));
+            return new Value(destTy, reg);
+        }
+        else {
+            panic err:illegalArgument("value must be int type");
+        }
+    }
+
     public function unreachable() {
         BasicBlock bb = self.bb();
         bb.addInsn("unreachable");

--- a/compiler/modules/print.llvm/tests/intToFloat.bal
+++ b/compiler/modules/print.llvm/tests/intToFloat.bal
@@ -1,0 +1,22 @@
+import ballerina/test;
+
+function intToFloat() returns Module {
+    Context context = new;
+    Builder builder = context.createBuilder();
+    Module m = context.createModule();
+    FunctionDefn test = m.addFunctionDefn("test", {returnType: "double", paramTypes: ["i64", "i32"]});
+    BasicBlock initBlock = test.appendBasicBlock();
+    builder.positionAtEnd(initBlock);
+    Value p0 = test.getParam(0);
+    Value p1 = test.getParam(1);
+    Value v1 = builder.sIToFP(p0, "double");
+    Value v2 = builder.sIToFP(p1, "double");
+    Value ret = builder.fArithmetic("fadd", v1, v2);
+    builder.ret(ret);
+    return m;
+}
+
+@test:Config {}
+function testIntToFloat() returns error? {
+    return runTest(intToFloat, "int_to_float.ll");
+}

--- a/compiler/modules/print.llvm/tests/testOutputs/int_to_float.ll
+++ b/compiler/modules/print.llvm/tests/testOutputs/int_to_float.ll
@@ -1,0 +1,6 @@
+define double @test(i64 %0, i32 %1) {
+  %3 = sitofp i64 %0 to double
+  %4 = sitofp i32 %1 to double
+  %5 = fadd double %3, %4
+  ret double %5
+}

--- a/llvm_jni/modules/jni.llvm/Builder.bal
+++ b/llvm_jni/modules/jni.llvm/Builder.bal
@@ -248,6 +248,11 @@ public distinct class Builder {
         return new (jLLVMBuildAddrSpaceCast(self.LLVMBuilder, val.LLVMValueRef, typeToLLVMType(destTy), java:fromString(reg)));
     }
 
+    public function sIToFP(Value val, RealType destTy, string? name=()) returns Value {
+        string reg = self.extractName(name);
+        return new (jLLVMBuildSIToFP(self.LLVMBuilder, val.LLVMValueRef, typeToLLVMType(destTy), java:fromString(reg)));
+    }
+
     function extractName(string? name) returns string {
         if name is () {
             return "";
@@ -497,6 +502,12 @@ function jLLVMBuildFNeg(handle builder, handle val, handle name) returns handle 
     name: "LLVMBuildFNeg",
     'class: "org.bytedeco.llvm.global.LLVM",
     paramTypes: ["org.bytedeco.llvm.LLVM.LLVMBuilderRef", "org.bytedeco.llvm.LLVM.LLVMValueRef", "java.lang.String"]
+} external;
+
+function jLLVMBuildSIToFP(handle builder, handle val, handle destTy, handle name) returns handle = @java:Method {
+    name: "LLVMBuildSIToFP",
+    'class: "org.bytedeco.llvm.global.LLVM",
+    paramTypes: ["org.bytedeco.llvm.LLVM.LLVMBuilderRef", "org.bytedeco.llvm.LLVM.LLVMValueRef", "org.bytedeco.llvm.LLVM.LLVMTypeRef", "java.lang.String"]
 } external;
 
 function jLLVMSetAlignment(handle valueRef, int bytes) = @java:Method {

--- a/llvm_jni/modules/jni.llvm/tests/intToFloat.bal
+++ b/llvm_jni/modules/jni.llvm/tests/intToFloat.bal
@@ -1,0 +1,22 @@
+import ballerina/test;
+
+function intToFloat() returns Module {
+    Context context = new;
+    Builder builder = context.createBuilder();
+    Module m = context.createModule();
+    FunctionDefn test = m.addFunctionDefn("test", {returnType: "double", paramTypes: ["i64", "i32"]});
+    BasicBlock initBlock = test.appendBasicBlock();
+    builder.positionAtEnd(initBlock);
+    Value p0 = test.getParam(0);
+    Value p1 = test.getParam(1);
+    Value v1 = builder.sIToFP(p0, "double");
+    Value v2 = builder.sIToFP(p1, "double");
+    Value ret = builder.fArithmetic("fadd", v1, v2);
+    builder.ret(ret);
+    return m;
+}
+
+@test:Config {}
+function testIntToFloat() returns error? {
+    return runTest(intToFloat, "int_to_float.ll");
+}

--- a/llvm_jni/modules/jni.llvm/tests/testOutputs/int_to_float.ll
+++ b/llvm_jni/modules/jni.llvm/tests/testOutputs/int_to_float.ll
@@ -1,0 +1,6 @@
+define double @test(i64 %0, i32 %1) {
+  %3 = sitofp i64 %0 to double
+  %4 = sitofp i32 %1 to double
+  %5 = fadd double %3, %4
+  ret double %5
+}

